godot-core/src/obj/onready.rs

```diff
@@ -65,10 +75,77 @@ use std::mem;
 ///        assert_eq!(*self.manual, 22);
 ///     }
 /// }
+/// ```
+///
+/// # Example - Macro-generated `init`
+/// ```
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct MyClass {
+///    base: Base<Node>,
+///    #[init(default = OnReady::new(|| 11))]
+///    auto: OnReady<i32>,
+///    #[init(default = OnReady::manual())]
+///    manual: OnReady<i32>,
+///    #[init(node = "ChildPath")]
+///    auto_node: OnReady<Gd<Node2D>>,
+/// }
+///
+/// #[godot_api]
+/// impl INode for MyClass {
+///     fn ready(&mut self) {
+///        // self.auto is now ready with value 11.
+///        assert_eq!(*self.auto, 11);
+///
+///        // self.manual needs to be initialized manually.
+///        self.manual.init(22);
+///        assert_eq!(*self.manual, 22);
+///
+///        // self.auto_node is now ready with the node found at path `ChildPath`
+///        assert_eq!(self.auto_node.get_name(), "ChildPath".into());
+///     }
+/// }
+/// ```
 pub struct OnReady<T> {
     state: InitState<T>,
 }
 
+impl<T: GodotClass + Inherits<Node>> OnReady<Gd<T>> {
+    /// Variant of [OnReady::new], except this is hard-coded to fetch the node at `path` before ready.
```

Comment:
```suggestion
    /// Variant of [`OnReady::new()`], fetching the node located at `path` before `ready()`.
```
Also other occurrences